config/config.go

```diff
@@ -90,56 +90,83 @@ func parsePlugins(pluginList []interface{}) ([]*PluginConfig, error) {
 	return plugins, nil
 }
 
-func (c *Config) parseV6Config() error {
-	if exists := c.v.Get("server6"); exists == nil {
-		// it is valid to have no DHCPv6 configuration defined, so no
-		// server and no error are returned
-		return nil
+func (c *Config) getListenAddress(v6 bool) (*net.UDPAddr, error) {
```

Comment:
Is a bool the right argument type here? Bool args tend to be uninformative at the call site.  What about instead something like:

```go
type configFamily int // or some similar type name

const (
    familyV4 configFamily = 4
    familyV6 configFamily = 6
)

func (c *Config) getListenAddress(family configFamily) (*net.UDPAddr, error) {}
```

Then you call with:

```go
c.getListenAddress(familyV4)
```

Then you can then get rid of all the `ver` declarations and `if` statements at function start below, and just use the `configFamily` directly (or assert as int if necessary).

Reply:
I had the same doubt at first, but yeah, this would be much better, I'll change it